plugins/report-portal/src/api/ReportPortalClient.ts

```diff
@@ -0,0 +1,66 @@
+import { DiscoveryApi } from '@backstage/core-plugin-api';
+
+import { ReportPortalApi } from './ReportPortalApi';
+import { LaunchDetailsResp, ProjectDetails, ProjectListResp } from './types';
+
+export class ReportPortalClient implements ReportPortalApi {
+  constructor(private readonly discoveryApi: DiscoveryApi) {}
+
+  private async getBaseApiUrl() {
+    return `${await this.discoveryApi.getBaseUrl('report-portal')}/v1/`;
+  }
+
+  getReportPortalBaseUrl(host: string) {
+    return `https://${host}/`;
+  }
+
+  async getLaunchResults(
+    projectId: string,
+    host: string,
+    filters: { [key: string]: string | number } | undefined,
+  ) {
+    const baseUrl = new URL(
+      `${projectId}/launch/latest`,
+      await this.getBaseApiUrl(),
+    );
+    if (filters) {
+      Object.keys(filters).forEach(key =>
+        baseUrl.searchParams.append(key, filters[key] as string),
+      );
+    }
+    baseUrl.searchParams.append('host', host);
+    const response = await fetch(baseUrl);
+    if (response.status >= 400 && response.status < 600) {
+      throw new Error('Failed to fetch launch details');
+    }
+    return (await response.json()) as LaunchDetailsResp;
+  }
+
+  async getProjectDetails(projectId: string, host: string) {
+    const baseUrl = new URL(`project/${projectId}`, await this.getBaseApiUrl());
+    baseUrl.searchParams.append('host', host);
+    const response = await fetch(baseUrl);
+    if (response.status >= 400 && response.status < 600) {
```

Comment:
make a condion of response.status != 200

Reply:
changed this condition